tests/testthat/test-revision-latency-functions.R

```diff
@@ -35,7 +35,84 @@ test_that("revision_summary works for a dummy dataset", {
   expect_snapshot(dummy_ex %>% revision_summary(drop_nas = FALSE) %>% print(n = 10, width = 300))
 })
 test_that("tidyselect is functional", {
-  expect_no_error(revision_summary(dummy_ex, value))
-  expect_no_error(revision_summary(dummy_ex, starts_with("val")))
+  expect_no_error(quiet(revision_summary(dummy_ex, value)))
+  expect_no_error(quiet(revision_summary(dummy_ex, starts_with("val"))))
+  with_later_key_col <- dummy_ex$DT %>%
+    select(geo_value, time_value, value, version) %>%
+    as_epi_archive(versions_end = dummy_ex$versions_end, compactify = FALSE)
+  expect_equal(
+    quiet(revision_summary(with_later_key_col)),
+    quiet(revision_summary(dummy_ex))
+  )
+  with_later_val_col <- dummy_ex$DT %>%
+    mutate(value2 = 0) %>%
+    as_epi_archive(versions_end = dummy_ex$versions_end, compactify = FALSE)
+  expect_equal(
+    quiet(revision_summary(with_later_val_col, value)),
+    quiet(revision_summary(dummy_ex, value))
+  )
+  expect_error(
+    dummy_ex$DT %>%
+      copy() %>%
+      mutate(value2 = value) %>%
+      as_epi_archive(
+        versions_end = dummy_ex$versions_end,
+        compactify = FALSE
+      ) %>%
+      revision_summary(),
+    class = "epiprocess__revision_summary_cannot_determine_default_selection"
+  )
+  expect_error(revision_summary(with_later_val_col, !everything()),
+    class = "epiprocess__revision_summary__selected_zero_columns"
+  )
+})
+
+test_that("revision_summary default min_waiting_period works as expected", {
+  expect_equal(
+    tibble(
+      geo_value = 1,
+      time_value = as.Date("2020-01-01") + 0:1,
+      version = time_value + 1,
+      value = 1:2
+    ) %>%
+      as_epi_archive(versions_end = as.Date("2020-01-01") + 1 + 59) %>%
+      revision_summary(print_inform = FALSE) %>%
+      pull(time_value),
+    as.Date("2020-01-01")
+  )
+  expect_equal(
```

Comment:
wasn't able to annotate this one, not quite sure what the difference is supposed to be.

Reply:
I guess this is weekly data?

Reply:
- [ ] add comments re. time_type